kernel/kernel-api/src/test/scala/io/delta/kernel/internal/columndefaults/ColumnDefaultsSuite.scala

```diff
@@ -0,0 +1,166 @@
+import io.delta.kernel.test.ActionUtils
+import io.delta.kernel.types._
+
+import org.scalatest.funsuite.AnyFunSuite
+
+class ColumnDefaultsSuite extends AnyFunSuite with ActionUtils {
+  val validProtocol = new Protocol(
+    TableFeatures.TABLE_FEATURES_MIN_READER_VERSION,
+    TableFeatures.TABLE_FEATURES_MIN_WRITER_VERSION,
+    Set.empty[String].asJava,
+    Set(
+      TableFeatures.ALLOW_COLUMN_DEFAULTS_W_FEATURE.featureName(),
+      TableFeatures.ICEBERG_COMPAT_V3_W_FEATURE.featureName()).asJava)
+  test("validate") {
+    def metadataForDefault(value: String): FieldMetadata =
+      FieldMetadata.builder().putString("CURRENT_DEFAULT", value).build()
+    val correctSchema = new StructType()
+      .add("id", IntegerType.INTEGER)
+      .add("int", IntegerType.INTEGER, metadataForDefault("\"123\""))
+      .add("short", ShortType.SHORT, metadataForDefault("123"))
+      .add("long", LongType.LONG, metadataForDefault("1231341"))
+      .add("decimal", new DecimalType(10, 5), metadataForDefault("1231.34155"))
+      .add("float", FloatType.FLOAT, metadataForDefault("123.1341"))
+      .add("double", DoubleType.DOUBLE, metadataForDefault("123.7774"))
+      .add("double2", DoubleType.DOUBLE, metadataForDefault("'123.7774'"))
+      .add("name", StringType.STRING, metadataForDefault("\"tom\""))
+      .add("name2", StringType.STRING, metadataForDefault("'tom'"))
+      .add("date", DateType.DATE, metadataForDefault("'2025-01-01'"))
+      .add("date2", DateType.DATE, metadataForDefault("2025-01-01"))
+      .add("ts", TimestampType.TIMESTAMP, metadataForDefault("\"2025-01-01T00:00:00Z\""))
+      .add("ts2", TimestampType.TIMESTAMP, metadataForDefault("\"2025-01-01T00:00:00+01:00\""))
+      .add("ts3", TimestampType.TIMESTAMP, metadataForDefault("2025-01-01T00:00:00Z"))
+      .add("tsntz", TimestampNTZType.TIMESTAMP_NTZ, metadataForDefault("'2025-01-01T00:00:00'"))
+      .add("tsntz2", TimestampNTZType.TIMESTAMP_NTZ, metadataForDefault("2025-01-01T00:00:00"))
+      .add(
+        "childStruct",
+        new StructType()
+          .add("childId", IntegerType.INTEGER, metadataForDefault("100"))
+          .add(
+            "grandChildList",
+            new ArrayType(
+              new StructType().add("nestedId", IntegerType.INTEGER, metadataForDefault("120")),
+              false)))
+      .add(
+        "grandChildMap",
+        new MapType(
+          new StructType().add("mapKeyId", IntegerType.INTEGER, metadataForDefault("220")),
+          new StructType().add("mapValueId", IntegerType.INTEGER, metadataForDefault("330")),
+          false))
+      .add(
+        "childList",
+        new ArrayType(
+          new StructType().add("clid", IntegerType.INTEGER, metadataForDefault("300")),
+          false))
+    ColumnDefaults.validateSchema(correctSchema, true, true)
+    val e = intercept[KernelException] {
+      ColumnDefaults.validateSchema(correctSchema, true, false)
```

Comment:
Test this case as well with a schema with no default values? Since that was the specific behavior we decided on